### PR TITLE
Shifting content to en/*.yaml files

### PR DESCRIPTION
**Pull request by czgdp1807**
#### Brief description of what is fixed or changed

- [x] `machine-learning.html`
- [x] `array-libraries.html`
- [x] `scientific-domains.html`
- [x] `data-science.html`
- [x] `visualization.html`
- [x] `shell-content.html`
- [x] `shell.html`

### CODE REVIEW

**Comment by czgdp1807** [comment]
cc @rgommers

**Comment by rgommers** [comment]
Thanks @czgdp1807. Looks like the right structure in the yaml file. The build isn't happy yet, but the approach looks good to me.

**Comment by czgdp1807** [comment]
Locally everything is working fine i.e., the site is building correctly.

**Comment by rgommers** [comment]
Ah okay - let me see if I can figure out what is going on with CI.

**Comment by czgdp1807** [comment]
> error calling index: index of untyped nil
​
This is the bug. Let me see if I can re-produce it locally.

**Comment by rgommers** [comment]
See https://gohugo.io/hosting-and-deployment/hosting-on-netlify/#configure-hugo-version-in-netlify. Can you add `HUGO_VERSION = "0.83.1"` under `[context.deploy-preview.environment]`?

**Comment by czgdp1807** [comment]
It seems like even after updating version things are not working on CI. I will try to fix this tomorrow.

**Comment by czgdp1807** [comment]
May be, [this](https://answers.netlify.com/t/what-am-i-doing-wrong-my-site-built-locally-with-hugo-looks-fine-but-there-are-problems-on-netlify/8560/5) can help.

**Comment by rgommers** [comment]
@czgdp1807 I can reproduce the error you're seeing in CI locally with Hugo `0.84.1+extended`. The thing is that by default the site doesn't build with translations enabled, because we haven't launched them yet. You can enable them with:
```diff
$ git diff
diff --git a/config.yaml.in b/config.yaml.in
index e10c6fd..3a5d379 100644
--- a/config.yaml.in
+++ b/config.yaml.in
@@ -12,7 +12,7 @@ markup:
 # Uncomment this to disable the translations for the specified languages.
 # This may be useful to be able to merge partial translations without deploying
 # them to numpy.org immediately.
-disableLanguages: ["pt", "ja"]
+#disableLanguages: ["pt", "ja"]
```

And then I see:
```bash
$ python gen_config.py 
$ hugo
Start building sites … 
hugo v0.84.1+extended linux/amd64 BuildDate=unknown
Total in 258 ms
Error: Error building site: failed to render pages: render of "home" failed: execute of template failed: template: index.html:19:7: executing "main" at <partial "tabs.html" .>: error calling partial: execute of template failed: template: partials/tabs.html:17:19: executing "partials/tabs.html" at <partial "array-libraries.html" .>: error calling partial: "/home/rgommers/code/numpy.org/layouts/partials/array-libraries.html:3:14": execute of template failed: template: partials/array-libraries.html:3:14: executing "partials/array-libraries.html" at <index $arraylibraries "intro">: error calling index: index of untyped nil
```
I'll have a look at updating the Hugo version to `0.84` anyway, but the current version will probably work fine too if you get to build the translations locally.

**Comment by rgommers** [comment]
Oh, an easier way to enable translations: `export NUMPYORG_WITH_TRANSLATIONS=1`

**Comment by rgommers** [comment]
Okay, Hugo version in GH Actions deploy job and in `netlify.toml` for the preview are in sync now, both `0.82.0-extended`.

File: config.yaml.in
Comment (czgdp1807) [inline config.yaml.in:38]: The fix happened after adding this line here. Should I use automatic translation to generate content for `pt/tabcontent.yaml` and `ja/tabscontent.yaml` or should I just leave the text empty with the keys added. As of now these files are absent.

Comment (rgommers) [reply config.yaml.in:38]: Ah that makes sense. I'd say just copy over the English `.yaml` file to `ja/` and `pt/`, that will then make the files available for translation after merging this PR.

File: layouts/partials/scientific-domains.html
Comment (czgdp1807) [inline layouts/partials/scientific-domains.html:73]: Do we need to shift these links to YAML files? I am asking because, I think even after translation to any language they will remain the same. The case with NumPy is the same and there is just link and library name here. In that case, it won't be much beneficial to have the same content repeated (in English) in all the yaml files under folders of different languages.

Comment (rgommers) [reply layouts/partials/scientific-domains.html:73]: Agreed, can stay here - names and links should not be translated

**Comment by czgdp1807** [comment]
@rgommers This is ready. Let me know if any changes are required. Thanks.

**Review by rgommers — APPROVED** [review]
I played with it locally, making changes works as expected. Looks excellent, aside from the one minor comment on not yet launched the translations yet. I'll fix that one line and will then merge. Thanks a lot @czgdp1807!

**Comment by rgommers** [comment]
@AtsushiSakai @kngwyu @melissawm the last parts are now translatable in the `pt/*.yaml` and `ja/*.yaml`. Apologies for the delay!

**Comment by AtsushiSakai** [comment]
I have translated remaining sentences in Crowdin.

**Comment by rgommers** [comment]
> I have translated remaining sentences in Crowdin.

Thanks @AtsushiSakai! I'm not sure I'm seeing them there though, the `config.yaml` has very few changes:

image

**Comment by rgommers** [comment]
Oh no, now I understand my confusion @AtsushiSakai. Most of the to-be-translated strings in this PR are in a new file `tabcontents.yaml`, and that file was not syncing to Crowdin. I'm trying to force it to update now.

**Comment by rgommers** [comment]
@Qiyu8 I just noticed you also did a ton of work on the Chinese translation, thank you! It looks to me like it could go into the initial launch together with Japanese and Portuguese. I will work on including the translations in the repo and making them show up in the PR preview now.

**Comment by rgommers** [comment]
`tabcontents.yaml` has now shown up in Crowdin.

**Comment by AtsushiSakai** [comment]
Sorry, I missed some sentences. I have translated remained sentences. PTAL.

**Comment by rgommers** [comment]
Awesome, thanks @AtsushiSakai! Back at 100%:) I'll wait for @kngwyu's approval.

**Comment by kngwyu** [comment]
Thanks! I'll try to check the new translations this week, hopefully.

**Comment by Qiyu8** [comment]
@rgommers Thanks! I have translated all pages into Chinese in Crowdin except for `tabcontents.yaml` and `config.yaml`. I will try to finish the translation in Crowdin this week.

**Pull request merged**
